src/main/groovy/com/google/protobuf/gradle/Utils.groovy

```diff
@@ -129,4 +133,78 @@ class Utils {
       }
     }
   }
+
+  /**
+   * Add the the folder of generated source to Eclipse .classpath file.
+   */
+  static void addToEclipseSources(Project project, boolean isTest, String sourceSetName, File f) {
+    project.plugins.withId("eclipse") {
+      File projectDir = project.getProjectDir()
+
+      EclipseModel model = project.getExtensions().findByType(EclipseModel)
+      model.classpath.file.whenMerged { Classpath cp ->
+        // buildship requires the folder exists on disk, otherwise
+        // it will be ignored when updating classpath file, see:
+        // https://github.com/eclipse/buildship/issues/1196
+        f.mkdirs()
+
+        String relativePath = projectDir.toURI().relativize(f.toURI()).getPath()
+        // remove trailing slash
+        if (relativePath.endsWith("/")) {
+          relativePath = relativePath[0..(relativePath.length() - 2)]
+        }
+        SourceFolder entry = new SourceFolder(relativePath, getOutputPath(cp, sourceSetName))
+        entry.entryAttributes.put("optional", "true")
+        entry.entryAttributes.put("ignore_optional_problems", "true")
+
+        // this attribute is optional, but it could be useful for IDEs to recognize that it is
+        // generated source folder from protobuf, thus providing some support for that.
+        // e.g. Hint user to run generate tasks if the folder is empty or does not exist.
+        entry.entryAttributes.put("protobuf_generated_source", "true")
+
+        // check if output is not null here because test source folder
+        // must have a separate output folder in Eclipse
+        if (entry.output != null && isTest) {
+          entry.entryAttributes.put("test", "true")
+        }
+        cp.entries.add(entry)
+      }
+    }
+  }
+
+  /**
+   * Get the output path according to the source set name, if no valid path can be
+   * found, <code>null</code> will return.
+   */
+  private static String getOutputPath(Classpath classpath, String sourceSetName) {
+    String path = "bin/" + sourceSetName
```

Comment:
This is the output path of what? The compiled classes output? I don't see it asserted in the test either.

Reply:
Yes it's the compiled classes output.

Buildship(Eclipse Gradle project importer) by default will set the output folder to `bin`